pkg/golinters/forbidigo.go

```diff
@@ -40,14 +41,20 @@ func NewForbidigo(settings *config.ForbidigoSettings) *goanalysis.Linter {
 		},
 	}
 
+	loadMode := goanalysis.LoadModeSyntax
+	if settings != nil && settings.ExpandExpressions {
```

Comment:
I believe we also need to set the analyzer flag called `analyze_types` when we see this set.

Reply:
Which analyzer flag?

goanalysis uses different "levels", not individual flags: https://pkg.go.dev/github.com/golangci/golangci-lint/pkg/golinters/goanalysis#LoadMode

Reply:
I was thinking we were using forbidigo as an analyzer, per the new guidelines for linters.  For an analyzer, we'd set the `analyze_types` flag at https://github.com/ashanbrown/forbidigo/blob/master/pkg/analyzer/analyzer.go#L55.  That said, it looks like we've set up forbidigo to automatically expand types when `TypesInfo` is provided. I think I hadn't quite realized we were using that as a flag, rather than just information.  In retrospect, my preference would have been to explicitly add a new `analyzeTypes` option at https://github.com/ashanbrown/forbidigo/blob/master/forbidigo/forbidigo.go#L65 (similar for what we did for the analyzer mode).

Reply:
By using types, your linter is now considered a slow linter.
https://github.com/golangci/golangci-lint/pull/3639

Reply:
@ldez That makes sense, and we had anticipated this issue, although I didn't realize that it was formally checked in golangci-lint.  The linter can work without types, but there is an opt-in behavior that allows us to use types to match more things.  Is there any precedent for a linter being both fast and slow?  I don't suppose we could have both fast and slow versions of the same linter?  I'd imagine they would have to have separate names and configurations.

Reply:
A linter cannot be slow and fast, because the slow part is the type analysis, and the type analysis is not an option handled by the linter itself.
I don't think that duplicating a linter is a solution.
Your linter will just be "slow" now.

Reply:
Is type analysis triggered by the line at https://github.com/golangci/golangci-lint/pull/3612/files#diff-b647b23fb0e6073b7f161d88a8cd16af6dd3e90de1d6a81006d2d9a4ee416707R424?  In this PR, it appears that setting is conditionally executed based on that configuration.

Reply:
The goal was to not change behavior unless the user explicitly asks  for it in the config. It looks like there are two places where that is relevant:
- GetAllSupportedLinterConfigs: covered by this PR
- NewForbidigo: covered by this PR

The already [merged commit](https://github.com/golangci/golangci-lint/pull/3639/commits/06b681e96329e96779b008c301bc5f9460b6c111) unconditionally reconfigures forbidigo to enable type analysis. There's a certain risk that this breaks users because existing patterns get interpreted differently. I missed earlier that the PR wasn't just the automatic bump of the forbidigo package.

I think we should merge this PR here soon because it is required to finish the forbidigo update to 1.5.x properly.

Reply:
Wait, I'm not keeping up with you guys :cry: 

So forbidigo 1.5.x is new, and the upgrade of it in https://github.com/golangci/golangci-lint/pull/3639 is something that I really only saw today because it's also new.

Reply:
That PR makes forbidigo a "slow" linter, without actually enabling the feature which uses the additional information (no call of `OptionAnalyzeTypes(true)`.

So my statement above still stands: this PR is needed to finish the forbidigo update. I just need to add `OptionAnalyzeTypes(true)`.

Reply:
I've added it.